source/agora/node/Ledger.d

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+
+    Contains supporting code for tracking the current ledger.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.Ledger;
+
+import agora.common.Block;
+import agora.common.Data;
+import agora.common.Hash;
+import agora.common.Transaction;
+
+/// Ditto
+class Ledger
+{
+    /// data storage for all the blocks
+    private Block[Hash] ledger;
+
+    /// latest block height
+    private Block last_block;
```

Comment:
Pointer ?

Reply:
That's an extra dereference I guess, but we're fine with it?

Reply:
I take a dereference over a copy any time of the day.